### DISCUSSION TITLE
Select and paste buffer

### DISCUSSION THREAD

**Discussion by ghost** [kovidgoyal/kitty#4857] [Q&A] [answered]
Hello,

Is it possible to set buffers (for example in kitty.config) and select and paste them into a terminal from a dropdown menu (or similar)?

I'm trying to move away from TMUX and this is a feature I need, since the machines I connect via SSH/Telnet don't support of functions/aliases.

Basically my goal is to save sequences of commands I use frequently and paste them in the terminal when I need to run any of them.

**Comment by page-down** [comment, marked as answer]
> ... save sequences of commands I use frequently and paste them in terminal ...

I think you can write a kitten to do this. Similar to `insert password` in the link below.

https://sw.kovidgoyal.net/kitty/kittens/custom/

Drop-down menus require a mechanism similar to floating windows, which is not available yet, but you can use overlay.

**Comment by kovidgoyal** [comment]
There's no menu for it, but you can setup buffers mapped to key presses
already: https://sw.kovidgoyal.net/kitty/conf/#opt-kitty.copy_on_select
